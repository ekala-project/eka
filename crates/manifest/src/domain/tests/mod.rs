use super::*;
use toml_edit::de::from_str;

#[test]
fn depend() -> anyhow::Result<()> {
    let atom_str = r#"
            trait = "package"
            [atom]
            id = "foo"
            version = "0.1.0"

            [deps.foo]
            version = "^1"
            repo = "https://example.com/foo/bar.git"
        "#;

    insta::assert_yaml_snapshot!([from_str::<Depend>(atom_str)?]);

    Ok(())
}
