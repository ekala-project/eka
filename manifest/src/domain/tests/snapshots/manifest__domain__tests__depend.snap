---
source: manifest/src/domain/tests/mod.rs
expression: "[from_str::<Depend>(atom_str)?]"
---
- trait: package
  atom:
    id: foo
    version: 0.1.0
  deps:
    foo:
      version: ^1
      repo: "https://example.com/foo/bar.git"
